### PR TITLE
[breaking-change] Updated `Checkbox.fillColor` behavior

### PR DESCRIPTION
**Pull request by TahaTesser**
Migration guide for https://github.com/flutter/flutter/pull/125643

fixes https://github.com/flutter/flutter/issues/130295

## Presubmit checklist

- [x] This PR doesn’t contain automatically generated corrections (Grammarly or similar).
- [x] This PR follows the [Google Developer Documentation Style Guidelines](https://developers.google.com/style) — for example, it doesn’t use _i.e._ or _e.g._, and it avoids _I_ and _we_ (first person).
- [x] This PR uses [semantic line breaks](https://github.com/dart-lang/site-shared/blob/main/doc/writing-for-dart-and-flutter-websites.md#semantic-line-breaks) of 80 characters or fewer.

### CODE REVIEW

File: src/release/breaking-changes/checkbox-fillColor.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+border uses the `Checkbox.side` color when it is unchecked.
+
+## Description of change
+
+The `Checkbox.fillColor` is now applied to the checkbox's background when
+it is unchecked instead of the border.
+
+## Migration guide
+
+Updated `Checkbox.fillColor` behavior applies the fill color to the
```
Comment (HansMuller) [inline src/release/breaking-changes/checkbox-fillColor.md:27]:
```suggestion
The updated `Checkbox.fillColor` behavior applies the fill color to the
```

File: src/release/breaking-changes/checkbox-fillColor.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+title: Updated `Checkbox.fillColor` behavior
+description: >
+  Improved `Checkbox.fillColor` behavior applies the fill color to the
+  checkbox's background when it is unchecked.
+---
+
+## Summary
+
+The `Checkbox.fillColor` is now applied to the checkbox's background when
+it is unchecked.
```
Comment (HansMuller) [inline src/release/breaking-changes/checkbox-fillColor.md:11]:
```suggestion
the checkbox is unselected.
```

File: src/release/breaking-changes/checkbox-fillColor.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+## Context
+
+Previously, the `Checkbox.fillColor` was applied to the checkbox's border
+when it is unchecked and the background was transparent. With this change,
```
Comment (HansMuller) [inline src/release/breaking-changes/checkbox-fillColor.md:16]:
```suggestion
when the checkbox was unselected and its background was transparent. With this change,
```

**Review by HansMuller — APPROVED** [review]
LGTM

**Review by parlough — APPROVED** [review]
Looks good to me!

Thanks so much for submitting this migration guide. It's really appreciated.

In the future, feel free to open the PR from a branch in the repository for automatic staging.

**Pull request merged**